src/test/java/service/RoleServiceImplTest.java

```diff
@@ -0,0 +1,57 @@
+package service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import mate.academy.dao.RoleDao;
+import mate.academy.model.Role;
+import mate.academy.service.RoleService;
+import mate.academy.service.impl.RoleServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class RoleServiceImplTest {
+    private static RoleDao roleDao;
+    private static RoleService roleService;
+
+    @BeforeAll
+    static void beforeAll() {
+        roleDao = Mockito.mock(RoleDao.class);
+        roleService = new RoleServiceImpl(roleDao);
+    }
+
+    @Test
+    void save_Ok() {
+        Role expected = new Role(Role.RoleName.USER);
+        Mockito.when(roleDao.save(expected)).thenReturn(expected);
+        Role actual = roleService.save(expected);
+        assertEquals(expected.getRoleName().name(), actual.getRoleName().name());
+    }
+
+    @Test
+    void save_roleIsNull_NotOk() {
+        Mockito.when(roleDao.save(null)).thenReturn(null);
```

Comment:
I thint we should throw some exception here, not return null
<img width="602" alt="image" src="https://user-images.githubusercontent.com/68428976/186871255-6dec63b1-83ca-44d3-a5f8-b4d6fd8c2cce.png">